server/src-lib/Hasura/RQL/DDL/Schema/Rename.hs

```diff
@@ -348,6 +353,49 @@ updateColInRel fromQT rn rnCol = do
       updateColInArrRel fromQT toQT rnCol <$> decodeValue oldDefV
   liftTx $ updateRel fromQT rn newDefV
 
+updateColInRemoteRelationship
+  :: (MonadTx m, CacheRWM m)
+  => RemoteRelationshipName -> RenameCol -> m ()
+updateColInRemoteRelationship remoteRelationshipName renameCol = do
+  let (RenameItem qt oldCol newCol) = renameCol
+  (RemoteRelationshipDef remoteSchemaName hasuraFlds remoteFields) <-
+    liftTx $ RR.getRemoteRelDefFromCatalog remoteRelationshipName
+  let oldColPGTxt = getPGColTxt oldCol
+  let newColPGTxt = getPGColTxt newCol
+  let oldColFieldName = FieldName $ oldColPGTxt
+  let newColFieldName = FieldName $ newColPGTxt
+  let modifiedHasuraFlds = Set.insert newColFieldName $ Set.delete oldColFieldName hasuraFlds
+  let fieldCalls = unRemoteFields remoteFields
+  let oldColName = G.Name $ oldColPGTxt
+  let newColName = G.Name $ newColPGTxt
+  let modifiedFieldCalls = NE.map (\(FieldCall name args) ->
+                                     let remoteArgs = getRemoteArguments args
+                                     in FieldCall name $ RemoteArguments $
+                                         map (\(G.ObjectFieldG key val) ->
+                                                G.ObjectFieldG key $ replaceVariableName oldColName newColName val
+                                             ) $ remoteArgs
+                                  ) $ fieldCalls
+  RR.runUpdateRemoteRelationship (RemoteRelationship remoteRelationshipName qt modifiedHasuraFlds remoteSchemaName (RemoteFields modifiedFieldCalls))
```

Comment:
Use `liftTx $ RR.updateRemoteRelInCatalog (RemoteRelationship remoteRelationshipName ... ` to just update the catalog. This also let us to avoid `CacheRWM m` constraint for the function.

Reply:
Done!